ComposerChallenge/Views/ComposerView.swift

```diff
@@ -75,39 +105,58 @@ class ComposerView: UIView {
     // I'm leaving instructions to complete in each method to get them working
     
     // Constructing subviews means adding each view to the base view
-    // There is a method Apple proviedes that you can call directly to add subivews to the base view
+    // There is a method Apple provides that you can call directly to add subivews to the base view
     func constructSubviews() {
+        addSubview(composerImageView)
+        addSubview(nameLabel)
+        addSubview(bioLabel)
+        addSubview(alternateImageButton)
         
     }
     
     // This method will list all of the views you created above, allow them to use NSLayoutConstraints, then choose the NSLayoutConstraint you want for each view
     func constructSubviewConstraints() {
         // Create an array of your views
-
+        let views = [composerImageView, nameLabel, bioLabel, alternateImageButton]
         
-        // For each view, turn their property `.translatesAutoresizingMaskIntoConstraints` to be false
+     // TODO: - Can we replace the constraints below with:
+        // for view in views {
+        //    view.centerYAnchor.constraint(equalTo: self.centerYAnchor),
+        //    view.translatesAutoresizingMaskIntoConstraints = false,
+        //    ETC?
         
+        // For each view, turn their property `.translatesAutoresizingMaskIntoConstraints` to be false
+        composerImageView.translatesAutoresizingMaskIntoConstraints = false
+        nameLabel.translatesAutoresizingMaskIntoConstraints = false
+        bioLabel.translatesAutoresizingMaskIntoConstraints = false
+        alternateImageButton.translatesAutoresizingMaskIntoConstraints = false
         
         // Constrain each view
         
-        // composerImageView
+        
```

Comment:
If you constrain each view to `self.centerX` and `self.centerY`, they are all going to be centered and layered on top of each other. You might want to constrain each view to the bottom of the view above it etc.